src/app/pen.cpp

```diff
@@ -38,12 +39,21 @@ auto pen::process_events() -> event_loop_status
 
             // Move cursor to pen position, generate a click if the pen is
             // touching the screen
-            this->send_button_press(
-                screen_x, screen_y,
-                device_state.pressure > 0
-                    ? MouseButton::Left
-                    : MouseButton::None
-            );
+            MouseButton new_state =
+               device_state.pressure > 0
+                ? MouseButton::Left
+                : MouseButton::None;
+            this->send_button_press(screen_x, screen_y, new_state);
+
+            if (this->state != new_state) {
+                if (this->state == MouseButton::Left)
```

Comment:
I believe this should be changed to

```suggestion
                if (new_state == MouseButton::Left)
```

Otherwise the condition is inverted since `this->state` contains the previous state at this point.

Reply:
Well spotted out!

Fixed the commit.